src/modeler/AutoRest.Swagger/Validation/PatchBodyParametersSchemaValidation.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Properties;
+using AutoRest.Core.Logging;
+using AutoRest.Swagger.Model.Utilities;
+using System.Collections.Generic;
+using AutoRest.Swagger;
+using AutoRest.Swagger.Model;
+using System.Text.RegularExpressions;
+using System.Linq;
+using AutoRest.Swagger.Validation.Core;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class PatchBodyParametersSchemaValidation : TypedRule<Dictionary<string, Schema>>
+    {
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "M2016";
+
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.RPCViolation;
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => "PATCH operation '{0}' has reference to model definition '{1}' as request body parameter which has {2} properties";
```

Comment:
It's a bit tricky since I also need to make the message short so that it appears properly on the linter. Let me see what I can do.